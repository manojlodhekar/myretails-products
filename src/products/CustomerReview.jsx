import React from 'react';
import { getStarRating} from '../utils/ProductUtils';

const CustomerReview = (props) =>{
    
    let displayDate = new Date(props.customerReview.datePosted).toDateString().split(' ');
    displayDate = `${displayDate[1]} ${displayDate[2]}, ${displayDate[3]}`;

    return(
        <div>
            <div className="starCls"> {getStarRating(props.customerReview.overallRating)} </div>
            <div className="customerReviewTitle">{props.customerReview.title}</div>
            <div className="returnPolicyCls">{props.customerReview.review}</div>
            <div className="returnPolicyCls"><span style={{color: '#0066cc'}}>{props.customerReview.screenName}</span> {displayDate}</div>
        </div>
    )
}
export default CustomerReview;