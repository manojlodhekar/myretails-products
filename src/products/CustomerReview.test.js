import React from 'react';
import ReactDOM from 'react-dom';
import CustomerReview from './CustomerReview';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });
let customerReview;
const custReview = {
  "datePosted": "Mon Mar 11 13:13:55 UTC 2013",
  "overallRating": "1",
  "review": "Less than 2 months after purchase it completely stopped working. First it wouldn't detect the pitcher when trying to blend a significant amount, a couple weeks later it wouldn't detect the single serve cup. ",
  "reviewKey": "b326b0d6-e6ae-4ec5-8080-720f0ad741af",
  "screenName": "New York",
  "title": "Very unhappy"
 };

beforeEach(()=>{
  customerReview = mount(<CustomerReview customerReview={custReview}>
  </CustomerReview>);

})
describe('CustomerReview is defined', () => {

  it('should render correctly', () => {
    expect(customerReview).toHaveLength(1);
  });

  it('should render inner component', () => {
    expect(customerReview).toMatchSnapshot();
  });

  it('should render title of component', () => {
    customerReview = shallow(<CustomerReview customerReview={custReview}/>);
    expect(customerReview.find('div')).toHaveLength(5);
    expect(customerReview.find('.customerReviewTitle')).toHaveLength(1);
    expect(customerReview.find('.customerReviewTitle').text()).toEqual(custReview.title);
  });

  it('should render Review of component', () => {
    customerReview = shallow(<CustomerReview customerReview={custReview}/>);
     expect(customerReview.find('.returnPolicyCls')).toHaveLength(2);
    expect(customerReview.find('.returnPolicyCls').first().text()).toEqual(custReview.review);
  });

  it('should render datePosted of component', () => {
    customerReview = shallow(<CustomerReview customerReview={custReview}/>);
     expect(customerReview.find('.returnPolicyCls')).toHaveLength(2);
    expect(customerReview.find('.returnPolicyCls').last().text()).toEqual('New York Mar 11, 2013');
  });
  
 });
