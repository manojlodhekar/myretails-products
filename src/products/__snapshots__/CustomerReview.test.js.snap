// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerReview is defined should render inner component 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <CustomerReview
    customerReview={
        Object {
            "datePosted": "Mon Mar 11 13:13:55 UTC 2013",
            "overallRating": "1",
            "review": "Less than 2 months after purchase it completely stopped working. First it wouldn't detect the pitcher when trying to blend a significant amount, a couple weeks later it wouldn't detect the single serve cup. ",
            "reviewKey": "b326b0d6-e6ae-4ec5-8080-720f0ad741af",
            "screenName": "New York",
            "title": "Very unhappy",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
